File: app-react/src/pages/flutter/ui/Stepper.js
import React from 'react'
// ===================================================================== public component
import ListComponent from '@cpt/list.component'
// ===================================================================== component
const data = {
	title:'步骤指示器',
	grayText:'const',
	keyword:'Stepper( )',
	type:'Stepper',
	url:'StepperPage',
	data:[
		{ keyword:'steps', type:'List<Step>', pro:'步骤列表',
			code:[{path:'flutter/ui/Stepper/steps'}],
			link:[{title:'Step',url:'/flutter/style/Step'}]
		},
		{ keyword:'currentStep', type:'int', pro:'当前步骤' },
		{ keyword:'type', type:'StepperType', pro:'步骤类型', code:[{path:'flutter/ui/Stepper/StepperType'}] },
		{ keyword:'physics', type:'ScrollPhysics', pro:'响应用户输入', code:[{ path:'flutter/public/ScrollPhysics'}] },
		{ 
			keyword:'controlsBuilder', type:'Widget Function(BuildContext, {onStepContinue: void Function(), onStepCancel: void Function()})', pro:'控制器',
			code:[{path:'flutter/ui/Stepper/controlsBuilder'}]
		},
		{ keyword:'onStepTapped', type:'void Function( int )', pro:'点击步骤时执行', code:[ {path:'flutter/ui/Stepper/onStepTapped'} ] },
		{ keyword:'onStepContinue', type:'void Function( )', pro:'执行 controlsBuilder 中的 onStepContinue', code:[ {path:'flutter/ui/Stepper/onStepContinue'} ] },
		{ keyword:'onStepCancel', type:'void Function( )', pro:'执行 controlsBuilder 中的 onStepCancel', code:[ {path:'flutter/ui/Stepper/onStepCancel'} ] },
	]
}
export default class Index extends React.Component{ render(){ return <ListComponent data={data} {...this.props}/> } }

File: app-flutter/lib/pages/flutter/ui/Stepper.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 全局配置
import 'package:flutter_web_study.app/config.dart';
// ======================================================================================================== Widget 控件
class StepperPage extends StatefulWidget { // 有状态的控件，控件的状态、文字等属性存在变化的可能
    final String title;
    StepperPage({ Key key, this.title }) : super(key: key);
    @override
    _PageState createState() => _PageState(); // === createState()  => _PageState();
}

class _PageState extends State<StepperPage> {
    // page
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: Config.appBar(widget.title),
            body: ListView(
                padding:EdgeInsets.all(10.0),
                children: <Widget>[
                    Stepper(
                        type: StepperType.vertical,
                        steps: <Step>[
                            Step(
                                title: new Text('第一步'),
                                subtitle: new Text('complete'),
                                content: new Text('第一步内容'),
                                state: StepState.complete,
                                isActive: true,
                            ),
                            Step(
                                title: new Text('第二步'),
                                subtitle: new Text('disabled'),
                                isActive: true,
                                state: StepState.disabled,
                                content: new Text('第二步内容'),
                            ),
                            Step(
                                title: new Text('第三步'),
                                subtitle: new Text('error'),
                                state: StepState.error,
                                content: new Text('第三步内容'),
                            ),
                            Step(
                                title: new Text('第四步'),
                                subtitle: new Text('editing'),
                                state: StepState.editing,
                                content: new Text('第四步内容'),
                            ),
                            Step(
                                title: new Text('第五步'),
                                subtitle: new Text('indexed'),
                                state: StepState.indexed,
                                content: new Text('第五步内容'),
                            ),
                        ],
                        controlsBuilder:(BuildContext context, {VoidCallback onStepContinue, VoidCallback onStepCancel}) {
                            return Row(
                                children: <Widget>[
                                    FlatButton(
                                        onPressed: onStepContinue,
                                        child: const Text('继续'),
                                    ),
                                    FlatButton(
                                        onPressed: onStepCancel,
                                        child: const Text('取消'),
                                    ),
                                ],
                            );
                        },
                        onStepTapped: (int index){
                            print(index);
                        },
                        onStepContinue: (){
                            print('onStepContinue');
                        },
                        onStepCancel: (){
                            print('onStepCancel');
                        },
                    ),
                ],
            )
        );
    }
}